### PR TITLE
Implement CT "Review Article Journal"

### PR DESCRIPTION
**Pull request by thet**
Based on https://github.com/syslabcom/recensio.plone/pull/20

### CODE REVIEW

**Review by reinhardt — COMMENTED** [review]
The `editor` and `translatedTitle` fields are present in both `ReviewJournal` and `ReviewArticleJournal`. We could consider extracting them to a behavior.

NB. `ReviewJournal` uses the `title` field as the title of the reviewed journal. `ReviewArticleJournal` uses `titleJournal` for the title of the journal that contains the reviewed article. There are some shared semantics here but for migration purposes we should probably keep the fields as they are in the old system.

**Comment by ale-rt** [comment]
@reinhardt is there something that should be done before merging this PR? Should we merge this now and add another card?
@thet can you please rebase this dropping the commit that adds the base branch? It seems to me this can be merged separately?

**Comment by reinhardt** [comment]
> @reinhardt is there something that should be done before merging this PR? Should we merge this now and add another card?

No, nothing needs to be done right now.

I mentioned it because the Authors behavior also boils down to one field shared between two content types. If we want to be more consistent we could either make a behavior of `editor` and `translatedTitle`, or, if we want fewer behaviors, delete the Authors behavior and just add the `authors` field on the two individual types instead. Any of this can of course also be done later, or skipped altogether.

File: src/recensio/plone/profiles/default/types/Review_Article_Journal.xml [hunk 1]
```diff
@@ -48,6 +48,7 @@
     <element value="recensio.base_review" />
     <element value="recensio.cover_picture" />
     <element value="recensio.journal_review" />
```
Comment (ale-rt) [inline src/recensio/plone/profiles/default/types/Review_Article_Journal.xml:50]: @thet is there a reason why this has been added?

Comment (reinhardt) [reply src/recensio/plone/profiles/default/types/Review_Article_Journal.xml:50]: FWIW in the old code this content type is based on the `JournalReviewSchema` which is a combination of `PrintedReviewSchema` and some extra fields. See also #884

**Comment by reinhardt** [comment]
> the Authors behavior also boils down to one field shared between two content types

Sorry, that's not quite true - the authors schema is used by more content types (in the old system this happens via the `BookReviewSchema`).

**Pull request merged**